department-of-veterans-affairs/vets-website
# Add service number

## Summary
On the Pension Benefits application(21P-527EZ),
On the Step 2 Military History page

Add service number to Service Period

## How to run in local environment
1. Check out this branch locally
2. In your local environment, set the `pension_form_enabled` flipper to 'enabled' at http://localhost:3000/flipper/features/pension_form_enabled
3. Run `vets-website` and `vets-api`

## How to verify
Unauthenticated
1. Go to `http://localhost:3001/pension/application/527EZ` in your browser
2. Select 'Start your application without signing in'
3. Proceed to Step 2 by filling out the required fields on Step 1
4. On Step 2 - Verify service number field is included in Service Period
5. Verify service number is required and format is validated

Authenticated 
1. Sign in via `http://localhost:3001/sign-in/mocked-auth` to authenticate your user
2. Go to `http://localhost:3001/pension/application/527EZ` in your browser
3. Select Start application or continue application
4. Proceed to Step 2 by filling out the required fields on Step 1
5. On Step 2 - Verify service number field is included in Service Period
6. Verify service number is required and format is validated

## Related issue(s)
[Ticket in Github](https://github.com/department-of-veterans-affairs/va.gov-team/issues/68564)
[Epic in Github](https://github.com/department-of-veterans-affairs/va.gov-team/issues/63349)

## Testing done

- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots
_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

### Mobile
| Before | After | Required | Invalid | Valid |
| ------ | ------ | ------ | ------ | ------ |
|![localhost_3001_pension_application_527EZ_military_history(iPhone SE) (1)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/f2a19aa9-6c04-42ef-9888-6bc8adc3fb27)|![localhost_3001_pension_application_527EZ_military_history(iPhone SE)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/b03d5b93-165b-4262-b79d-0d23466b9222)|![localhost_3001_pension_application_527EZ_applicant_information(iPhone SE)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/131cbdbb-e4a9-4a1b-b18e-3477ad080c6e)|![localhost_3001_pension_application_527EZ_applicant_information(iPhone SE) (1)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/780b0c62-728c-430f-b55a-50ae6db1fcee)|![localhost_3001_pension_application_527EZ_applicant_information(iPhone SE) (2)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/18c2031d-d313-4f6e-baf5-e49a48ea978d)





### Desktop
| Before | After | Required | Invalid | Valid |
| ------ | ------ | ------ | ------ | ------ |
|![localhost_3001_pension_application_527EZ_military_history (1)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/09a298d8-a1c3-44ea-bd00-6706bbd693e9)|![localhost_3001_pension_application_527EZ_military_history](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/95b78ebd-38f3-4a7c-98e0-c6533cae8e3f)|![localhost_3001_pension_application_527EZ_applicant_information](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/47f5d1b0-945b-40be-9e24-3bc10b8d72ba)|![localhost_3001_pension_application_527EZ_applicant_information (1)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/89596ce4-ee70-4c84-8a1b-afa457fa829b)|![localhost_3001_pension_application_527EZ_applicant_information (2)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/b7b55777-cf64-4f5a-8906-e1ad2569b125)





## What areas of the site does it impact?
Pension Benefits Application

## Acceptance criteria
- On Step 2, Military history section, a "Service Number" field will be added as an additional question.
- It will be placed after the "Service end date" question and before "Add another service period" button.
- It will be labeled "Service number".
- Field will be required and requirement is shown to the user.
- If user chooses to enter an additional service period, they will have the same required question in the same placement.
- If the user does not enter anything within "Service Number" they will see an error that shows them it is a required field in order to save or continue.
- The “Service Number” field should have validation to ensure it follows the correct format.
- The field responses to this question should be saved so that it can be submitted.

### Quality Assurance & Testing

### Error Handling


### Authentication


### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:


## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
